### ISSUE TITLE
problem with deleting long discussion

### ISSUE THREAD

**Issue by zerzededd** [theahura/shoot-the-messenger#99]
Hey,

I have to restart deleting messages several times because after a while I get an error message.
I've already tried changing the rate limit with different values and it doesn't solve the problem.

Maybe I can have the extension retry automatically when it encounters this error. 

![image](https://github.com/theahura/shoot-the-messenger/assets/144461731/0f81b2ca-0a94-46ef-ad80-507d0b1eeea4)

**Comment by 8SecSleeper** [comment]
I plan on coming out of retirement to create another patch. Maybe this next
few days. I have that issue fixed in my test version.

**Comment by theahura** [comment]
@zerzededd can you post console logs? Not too easy to know what went wrong without more information

**Comment by theahura** [comment]
@8SecSleeper I can't replicate this by the way, but it seems to have popped up in a few of these bug reports. Do you know what this is?

**Comment by theahura** [comment]
Ok so I spent a solid chunk of today looking into this. 

It looks like facebook has modified how the search behavior works _again_. Previously we would call `element.form.dispatchEvent` in order to trigger search; now there is no parent form, so this returns an `undefined` error. 

I'm again pulling my hair trying to figure out how to simulate an enter click on the search bar :thinking:

**Comment by 8SecSleeper** [comment]
KeyboardEvent with DispatchEvent

**Comment by theahura** [comment]
Doesn't work, or at least I haven't been able to get it to work. 
Tried the following in the dev console
```
x = document.querySelectorAll('[placeholder="Search"]')[0]
z = new KeyboardEvent('keydown', {
    key: "Enter",
    code: "Enter",
    bubbles: true,
    cancelable: true,
    composed: true,
    defaultPrevented: true,
    keyCode: 13,
    returnValue: false,
    view: window,
    target: x,
    srcElement: x
})
x.dispatchEvent(z)
```
nothing happens on the actual page

**Comment by theahura** [comment]
Ok I think I got it. Man I had to go _deep_ into the guts of react for this one. 

TLDR: 
```
x.__reactFiber$frbte323rro.memoizedProps.children.props.children[0].props.children[0].props.commandConfigs[0].handler()
```
triggers the search event :grin: 

I think from here I can reverse engineer the rest

**Comment by theahura** [comment]
Alright, latest push should work. Closing this out
